### ISSUE TITLE
[bug] BG1 - Travel times sometimes differ 4h

### ISSUE THREAD

**Issue by caillean7** [gemrb/gemrb#590]
**Description**
When traveling via the worldmap between areas it sometimes takes 4 hours more in one direction. E. g. travel from the Friendly Arm Inn to Nashkel in 24 hours, but back in only 20 hours. From Nashkel mines to the carneval in 8 hours, back in 4. On the original it took 8 hours in both directions. Some routes seem okay, though. So it takes 12 hours to get from the Friendly Arm Inn to Beregost and vice versa.

**GemRB version**
0.86-git

**Expected behavior**
Travel times should be consistent in both directions

**Comment by lynxlynxlynx** [comment]
Assuming everything is flat, sure. I don't remember if the worldmap areas share links, but this makes it sound like they don't and the data is inconsistent. Sounds fishy.

**Comment by caillean7** [comment]
Didn't know that the travel times take the terrain into account (which makes sense), but I tried some routes in the original, and when they differ, it's always 4 hours. The one from Nashkel mines to the carnival really is the same on the IE while it differs on GemRB - should I test more, try to find more differences in travel times between the engines?

**Comment by lynxlynxlynx** [comment]
No, the one example is good enough. If NI has a less confusing worldmap explorer, you can look at the links between these two areas. DLTCEP suggests some of the exits have longer ways.

**Comment by caillean7** [comment]
This distance scaling? In the Nashkel example it's the same for both links. Carnival has but one south link, and the Nashkel mines one north link, both have distance scale 2. In NI you can have it display the traveling times for each area to its neighbours, btw. I'd say it's a little less confusing than DLTCEP, but honestly I have never looked at the worldmap editor before.

Found another discrepancy: From the Friendly Arm Inn to the first Cloakwood area (AR2200) the IE has 8 hours travel time, in the opposite direction 12 hours. GemRB has 12 hours in both directions. AR2200 has two east links with different scaling factors. But on both engines you arrive in the middle portion of the eastern edge of the map. This eastern Cloakwood exit connects to the southern Friendly Arm Inn one.

**Comment by fizzet** [comment]
This seems to be a case of the pathfinder being too smart. Taking the Nashkel example, going from the Carnival to the Mines takes 8 hours straight, but going via Nashkel town just takes 0 + 4...

Not sure what the right fix would be here. Always give priority to direct links?

**Comment by lynxlynxlynx** [comment]
I don't mind it being smarter, so we could just close this as wontfix. Making it consistent both ways would go against the data, so I'd leave that for mods.

But if we're more optimized, that doesn't explain the last (ar2200) example, so there probably is something to fix.

**Comment by fizzet** [comment]
It's most likely this
https://github.com/gemrb/gemrb/blob/dd96e92a3c18224ec66d66d50a3c41b3a75e7808/gemrb/core/WorldMap.cpp#L367-L368
Any idea why that comment reads like it does?

**Comment by lynxlynxlynx** [comment]
There was no test case mentioned when it was introduced in 7fce2b6b, so no idea.

**Comment by fizzet** [comment]
Reverting that commit fixes the discrepancy for AR2300 -> AR2200.

**Comment by lynxlynxlynx** [comment]
Ok, so the question just remains if that breaks anything else.
